pkg/dao/snapshot_model.go

```diff
@@ -0,0 +1,47 @@
+package dao
+
+import (
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+
+	"github.com/content-services/content-sources-backend/pkg/models"
+)
+
+type Snapshot struct {
+	models.Base
+	VersionHref      string `json:"version_href" gorm:"not null"`
+	PublicationHref  string `json:"publication_href" gorm:"not null"`
+	DistributionPath string `json:"distribution_path" gorm:"not null"`
+	DistributionHref string `json:"distribution_href" gorm:"not null"`
+	OrgId            string `json:"org_id" gorm:"not null"`
+	RepositoryUUID   string `json:"repository_uuid" gorm:"not null"`
+	Repository       models.Repository
+	ContentCounts    ContentCounts `json:"content_counts" gorm:"not null,default:{}"`
+}
+
+type ContentCounts map[string]int64
+
+func (cc *ContentCounts) Value() (driver.Value, error) {
+	if *cc == nil {
+		return "{}", nil
+	}
+	j, err := json.Marshal(cc)
+	return j, err
+}
+
+func (cc *ContentCounts) Scan(src interface{}) error {
+	source, ok := src.([]byte)
+	if !ok {
+		return errors.New("Type assertion .([]byte) failed.")
+	}
+
+	var counts ContentCounts
+	err := json.Unmarshal(source, &counts)
+	if err != nil {
+		return err
+	}
+
+	*cc = counts
+	return nil
+}
```

Comment:
note, that i moved the 'model' here.  I think longer term it makes sense to move all models to the dao layer, as it could reduce some of the boilerplate code.  We are using models really just as simple holders of data, so having it in a separate package isn't really providing much value, and i think making these more complex.  I'd consider this somewhat of a trial as we implement more, as i'm not 100% sold on it.  Curious what others think

Reply:
I agree. It makes `repository dao` in particular a lot more complex. The only packages that import `models` are `seeds` and `dao`, so it isn't even that big of a change. 

Would you also want to combine `seeds` in as well? IIRC `dao` tests imports seeds (for repository configs), so there would have to be some changes there to prevent a circular dependency.